tf_encrypted/protocol/securenn/securenn.py

```diff
@@ -353,9 +353,38 @@ def relu(self, x):
 
     :param PondTensor x: Input tensor.
     """
-    with tf.name_scope('relu'):
-      drelu = self.non_negative(x)
-      return drelu * x
+
+    def relu(x):
+      with tf.name_scope('relu'):
+        drelu = self.non_negative(x)
+        return drelu * x
+
+
+    max_size = 224*224*95
+    shape = x.shape.as_list()
+
+    if np.prod(shape) >= max_size:
+      #tensor is too big and will raise OOM
+
+      def check_num_split(num_split):
+        return (shape[-1] % num_split == 0 and
```

Comment:
maybe add a quick comment about what informally satisfies the predicate

Reply:
done